src/engine/controls/keycontrol.cpp

```diff
@@ -476,3 +525,106 @@ bool KeyControl::syncKey(EngineBuffer* pOtherEngineBuffer) {
     slotPitchChanged(pitchToTakeOctaves * 12);
     return true;
 }
+
+void KeyControl::slot432HzPitchLockChanged(double value) {
+    Q_UNUSED(value);
+    apply432HzPitchAdjustment();
+}
+
+void KeyControl::slotFileIs432HzChanged(double value) {
+    Q_UNUSED(value);
+    apply432HzPitchAdjustment();
+}
+
+void KeyControl::slotFileTuningFrequencyChanged(double value) {
+    Q_UNUSED(value);
+    apply432HzPitchAdjustment();
+}
+
+void KeyControl::ensureFileTuningControls() {
+    if (!m_pFileIs432Hz &&
+            ControlObject::exists(ConfigKey(getGroup(), "file_is_432hz"))) {
+        m_pFileIs432Hz = make_parented<ControlProxy>(
+                ConfigKey(getGroup(), "file_is_432hz"), this);
+        m_pFileIs432Hz->connectValueChanged(
+                this,
+                &KeyControl::slotFileIs432HzChanged,
+                Qt::DirectConnection);
+    }
+
+    if (!m_pFileTuningFrequencyHz &&
+            ControlObject::exists(ConfigKey(getGroup(), "file_tuning_frequency"))) {
+        m_pFileTuningFrequencyHz = make_parented<ControlProxy>(
+                ConfigKey(getGroup(), "file_tuning_frequency"), this);
+        m_pFileTuningFrequencyHz->connectValueChanged(
+                this,
+                &KeyControl::slotFileTuningFrequencyChanged,
+                Qt::DirectConnection);
+    }
+}
+
+bool KeyControl::fileMarkedAs432Hz() const {
+    if (m_pFileIs432Hz) {
+        return m_pFileIs432Hz->toBool();
+    }
+    return ControlObject::toBool(ConfigKey(getGroup(), "file_is_432hz"));
+}
+
+int KeyControl::detectedTuningFrequencyHz() const {
```

Comment:
This logic shall be moved into the Track object to just have a single double "file_key" we need to deal with.